Maven/tools/chocolateyUninstall.ps1

```diff
@@ -1,13 +1,16 @@
-$version = '3.6.1'
-$name = "apache-maven-$version"
-$tools = Split-Path $MyInvocation.MyCommand.Definition
-$package = Split-Path $tools
-$m2_home = Join-Path $package $name
-$mvn_cmd = Join-Path $m2_home 'bin/mvn.cmd'
-$mvn_debug_cmd = Join-Path $m2_home 'bin/mvnDebug.cmd'
+[Environment]::SetEnvironmentVariable('M2_HOME', $null, "Machine")
 
-[Environment]::SetEnvironmentVariable('M2_HOME', $null, "User")
+#Using registry method prevents expansion (and loss) of environment variables (whether the target of the removal or not)
+#To avoid bad situations - does not use substring matching or regular expressions
+#Removes duplicates of the target removal path, Cleans up double ";", Handles ending "\"
+$pathToRemove = '%M2_HOME%\bin'
+$unexpandedPath = (Get-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager\Environment" ).GetValue('Path', '', 'DoNotExpandEnvironmentNames') 
 
+foreach ($path in "$unexpandedPath".split(';')) {
+    If ($path -ine "$PathToRemove" -And $path -ine "$PathToRemove\") {
```

Comment:
Comparison seems inverted if I believe the [documentation](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_comparison_operators?view=powershell-6#-in) and we likely also need to revert the whole condition to only add entries from the PATH if they don't match the maven entry we're trying to remove:

```suggestion
    If -not ("$PathToRemove" -ine $path -Or"$PathToRemove\" -ine $path) {
```
What do you think?

Reply:
So, you got the intent of the code right. It is basically rebuilding the path with everything but the entry that we want to remove (as it is or suffixed by a forwarding slash).

`-ine` -> case insensitive not equal (same as `-ne` since Powershell is case insensitive by default). Not Equals is commutative - changing the order of the operands does not change the result.

By one of the [Demorgan's Laws](https://en.m.wikipedia.org/wiki/De_Morgan%27s_laws) Not (A or B) is the same as Not A and Not B.

We can express the conditionals as either:

    If ($path -ine "$PathToRemove" -And $path -ine "$PathToRemove\") {

Or: 

    If -not ($path -ieq "$PathToRemove" -Or $path -ieq "$PathToRemove") 

We can also use `-eq` / `-ne` (the `i` suffix is not necessary, it is just documenting the fact that I don't care about case).

Finally we can change the order of `$PathToRemove` and `$path` as per your example (but using `-eq` instead of `-ne`).

In your opinion, which variation is easier to read?

Reply:
You got the intent of the code right. It is basically rebuilding the path with everything but the entry that we want to remove (either as it is or suffixed by a forwarding slash).

`-ine` -> case insensitive not equal (same as `-ne` since Powershell is case insensitive by default). Not Equal is commutative - changing the order of the operands does not change the result.

By one of the [Demorgan's Laws](https://en.m.wikipedia.org/wiki/De_Morgan%27s_laws) Not (A or B) is the same as Not A and Not B.

We can express the conditionals as either:

    If ($path -ine "$PathToRemove" -And $path -ine "$PathToRemove\") {

Or: 

    If -not ($path -ieq "$PathToRemove" -Or $path -ieq "$PathToRemove") 

We can also use `-eq` / `-ne` (the `i` suffix is not necessary, it is just documenting the fact that we don't care about case sensitive).

Finally we can change the order of `$PathToRemove` and `$path` as per your example (but using `-eq` instead of `-ne`).

In your opinion, which variation is easier to read?

Reply:
Hi @MichaelPereira , sorry to bother you again. Could you have a look above? I've just figured that I've mixed camelCase with PascalCase, so I'm pushing some small changes, may as well change the If statement while on it.

Reply:
Hi @aaccioly, sorry for the delay, between the timezones and my personal email blocked at work, I missed the previous comment, so feel free to mention me, since that goes directly to my work email somehow :)

The first version of the conditional seems easier to maintain. I think we still need the two parts to be reversed since you're looking for a specific $pathToRemove in $path. I don't really know powershell that much but it looks like it needs to be written in this order according to the doc linked above.

Reply:
Hi @MichaelPereira, no worries as all. 

I've inverted the parts as requested. I guess that having `$pathToRemove` first is slightly more readable.

Just to clarify though:

    if($a -ine $b)

In PowerShell is equivalent to:

    if(!a.equalsIgnoreCase(b))

In Java. 

The order does not matter (well, in Java the order can matter if `a` or `b` is null). 
`-ine` is equivalent to `-ne` in the [documentation](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_comparison_operators?view=powershell-6) that you have linked.

If there is nothing else could you or @jpmat296  approve the PR (and potentially merge it to save me one step? :D).

Cheers,